opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java

```diff
@@ -255,16 +261,27 @@ public DiffData getDiffData() {
             InputStream[] in = new InputStream[2];
             try {
                 // Get input stream for both older and newer file.
+                ExecutorService executor = this.executor;
+                Future<?>[] future = new Future<?>[2];
                 for (int i = 0; i < 2; i++) {
                     File f = new File(srcRoot + filepath[i]);
-                    in[i] = HistoryGuru.getInstance().getRevision(f.getParent(), f.getName(), data.rev[i]);
+                    final String revision = data.rev[i];
+                    future[i] = executor.submit(() -> HistoryGuru.getInstance().
+                            getRevision(f.getParent(), f.getName(), revision));
+                }
+
+                for (int i = 0; i < 2; i++) {
+                    in[i] = (InputStream) future[i].get();
                     if (in[i] == null) {
                         data.errorMsg = "Unable to get revision "
                                 + Util.htmlize(data.rev[i]) + " for file: "
                                 + Util.htmlize(getPath());
+                        executor.shutdownNow();
                         return data;
                     }
                 }
+                // The Executor used by given repository will enforce the timeout.
+                executor.awaitTermination(0, TimeUnit.SECONDS);
```

Comment:
I'm not sure that a `shutdownNow()`/`awaitTermination()` is apt for a shared, long-running thread pool during a user request.

Reply:
oops. Thanks for spotting this.